co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManager.java

```diff
@@ -135,24 +151,99 @@ public V2xMessageTransmission onV2XMessageTx(InetAddress sourceAddr, CarmaV2xMes
     }
 
     /**
-     * Callback to be invoked when CARMA Platform receives a V2X Message from the NS-3 simulation
-     * @param rxMsg The V2X Message received
+     * Callback to be invoked when an RSU receives a V2X Message from the NS-3
+     * simulation
+     * 
+     * @param rxMsg   The V2X Message received
      * @param rxRsuId The Host ID of the vehicle receiving the data
-     * @throws RuntimeException If the socket used to communicate with the platform experiences failure
+     * @throws RuntimeException If the socket used to communicate with the platform
+     *                          experiences failure
      */
     public void onV2XMessageRx(byte[] rxMsg, String rxRsuId) {
-        if (!managedInstances.containsKey(rxRsuId))  {
+        if (!managedInstances.containsKey(rxRsuId)) {
             return;
         }
 
         InfrastructureInstance rsu = managedInstances.get(rxRsuId);
         try {
-            rsu.sendMsgs(rxMsg);
+            rsu.sendV2xMsg(rxMsg);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * Callback to be invoked when a infrastructure instance receives a simulated
+     * object detection from a registered simulated sensors. Only send object detection
+     * to the infrastructure instance that contains the sensor that reported it.
+     * 
+     * @param detection Detected Object.
+     */
+    public void onDetectedObject(DetectedObject detection) {
+        for (InfrastructureInstance instance : managedInstances.values()) {
+            if (instance.containsSensor(detection.getSensorId())) {
+                try {
+                    instance.sendInteraction(encodeObjectDetection(detection));
+                    // Assuming each sensor would only ever be registered to a single infrastructure
+                    // instance
+                    break;
+                } catch (IOException e) {
+                    log.error("Error occured:  {}", e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Helper method to serialize time sychronization message to JSON and encode into bytes 
+     * for eventual transmission via a Datagramsocket.
+     * 
+     * @param message time synchronization message to encode
+     * @return resulting bytes encoded from JSON string.
+     */
+    private byte[] encodeTimeMessage(InfrastructureTimeMessage message) {
```

Comment:
Feel like we can combine encodeTimeMessage and asJon into one function such as encodeMessage. Then replace all encode* functions with the generic encodeMessage function.

Reply:
No each send message sends to a different port.